### ISSUE TITLE
Blacklisting variables?

### ISSUE THREAD

**Issue by tsnieman** [apollographql/apollo-engine-js#22]
Is there a way to blacklist particular query/mutation variables from collection?

FWIW Optics has a very similar issue: https://github.com/apollographql/optics-agent-js/pull/126

**Comment by thepwagner** [comment]
`noTraceVariables` allows disabling _all_ variables.

We don't have a way to disable specific variables, but it's on the road map.

**Comment by pcarrier** [comment]
Would you rather pick variable names, or protect by type name (eg a custom `PIIString` scalar in your codebase)?

**Comment by tsnieman** [comment]
I'd prefer to be able to do something like this:

```
  engineConfig: {
    reporting: {
      // TODO https://github.com/apollographql/apollo-engine-js/issues/22
      // TODO blacklist individual variables
      // noTraceVariables: process.env.NODE_ENV === 'production',
      blacklistVariables: [
        'password',
        'anotherSecret',
        // etc.
      ],
    },
  }
```

Though I'll admit that I have no idea what this means:
> a custom `PIIString` scalar in your codebase

**Comment by thepwagner** [comment]
This is added in https://www.apollographql.com/docs/engine/proxy-release-notes.html#2017-11-121-g2a0310e1b-2017-11-30

Released in `apollo-engine-0.6.0`.

@tsnieman we were pretty close:
```
engineConfig: {
  reporting: {
    privateVariables: ['password','anotherSecret']
  }
}
```
(there was an existing option called `privateHeaders` so we kept naming aligned)
